### PR TITLE
Cleanup timestamp handling in vidl_ffmpeg source

### PR DESCRIPTION
**Pull request by mattdawkins**
Timestamps were previously not being populated to downstream processes

### CODE REVIEW

**Comment by dstoup** [comment]
Jenkins test this please

File: arrows/vxl/vidl_ffmpeg_video_input.cxx [hunk 1]
```diff
@@ -373,8 +380,42 @@ class vidl_ffmpeg_video_input::priv
...
+  bool presentation_time()
+  {
+    int frame_count( c_time_scan_frame_limit );
+
+    meta_ts = 0;
+    d_have_abs_frame_time = true;
+
+    while ( ( meta_ts == 0.0 )
+            && d_video_stream.advance()
+            && (( c_time_scan_frame_limit == 0) || frame_count-- ));
```
Comment (KyleFromKitware) [inline arrows/vxl/vidl_ffmpeg_video_input.cxx]: What exactly does `presentation_time()` represent? Does it just mean "start at timestamp 0"? What's the purpose of this bit which advances `d_video_stream`?

Comment (mattdawkins) [reply arrows/vxl/vidl_ffmpeg_video_input.cxx]: Yes, start at 0.

Not sure about the bit that advances d_video_stream, I copied it from other sections thinking it was handling scanning to a certain frame in the video but need to check (maybe it's redundant)

Comment (mleotta) [reply arrows/vxl/vidl_ffmpeg_video_input.cxx]: You don't want that while loop here.  If you look at the other cases, it's actually a do-while loop.  It's scanning the first `c_time_scan_frame_limit` frames looking for metadata with a timestamp.  If you don't need to scan, then please don't.

I thought "presentation time" means to use the FFMPEG presentation time variable, which doesn't always start at zero?

File: arrows/vxl/vidl_ffmpeg_video_input.cxx [hunk 2]
```diff
@@ -373,8 +380,42 @@ class vidl_ffmpeg_video_input::priv
...
+    using namespace std::chrono;
+    auto current_time = system_clock::now();
+    auto current_time_ms = time_point_cast<microseconds>(current_time);
+
+    meta_ts = current_time_ms.time_since_epoch().count();
+
+    d_have_abs_frame_time = true;
 
-// ------------------------------------------------------------------
+    while ( ( meta_ts == 0.0 )
```
Comment (mleotta) [inline arrows/vxl/vidl_ffmpeg_video_input.cxx]: This loop is not doing anything except slowing down your program when you open a video.

Comment (mattdawkins) [reply arrows/vxl/vidl_ffmpeg_video_input.cxx]: I'll remove it

File: arrows/vxl/vidl_ffmpeg_video_input.cxx (hunk 1 above)
Comment (mattdawkins) [reply arrows/vxl/vidl_ffmpeg_video_input.cxx]: I thought it was a relative time starting from 0, but I'm not well versed in mpg encoding. If you want I can change "presentation" to "start_at_0" to clean up any ambiguity?

**Comment by mattdawkins** [comment]
Jenkins test this please

**Pull request merged**